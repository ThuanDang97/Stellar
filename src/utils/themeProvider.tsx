import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
