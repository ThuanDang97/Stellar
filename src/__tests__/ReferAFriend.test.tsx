// Pages
import ReferAFriendPage from '@pages/refer-a-friend'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

describe('ReferAFriendPage Component', () => {
  test('Component matches DOM Snapshot', () => {
    const { container } = renderWithTheme(<ReferAFriendPage />)

    expect(container).toMatchSnapshot()
  })
})
