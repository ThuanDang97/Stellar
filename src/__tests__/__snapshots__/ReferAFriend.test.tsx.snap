// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReferAFriendPage Component Component matches DOM Snapshot 1`] = `
<div>
  <div
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__WrapperContent-sc-rqe5sa-6 klHjBi gVWyht"
  >
    <h1
      class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 gwtqfM ONSni"
      color="#242A30"
      font-family="AdobeCleanExtraBold"
      font-size="36"
      letter-spacing="2"
    >
      Refer A Friend Page
    </h1>
  </div>
</div>
`;
